### ISSUE TITLE
Keyboard navigation of cloud accounts misbehaves weirdly

### ISSUE THREAD

**Issue by nvdaes** [vslavik/poedit#888]
Synchronizing (uploading and downloading) translations hosted in the cloud requires to select an account, for example Crowdin, and this is not accessible with keyboard.
This makes difficult to use this feature to screen reader users.

**Comment by nvdaes** [comment]
In case you are interested, cc: @cyrilleB79, @mdcurran

**Comment by vslavik** [comment]
Please read [this document](https://github.com/vslavik/poedit/blob/master/SUPPORT.md) (which GitHub asked you to do when you created this issue), follow the instructions therein and provide detailed, explanatory information including steps to reproduce. Thanks!

**Comment by nvdaes** [comment]
### Steps to reproduce

1. Open Poedit free version 3.6.1 (7518) on Windows 11 11 24H2 (10.0.26100) with NVDA screen reader, version 2025.1beta1 (Spanish
2. Press alt to open menu.
3. Go to configuration, Accounts tab.
4. Press tab twice.
5. Press arrows to select an account provider like Crowding

### Expected results

A control accessible with keyboard, such as a combo box like other ones shown in Poedit gui, should be presented.

### Actual results

A control that NVDA screen reader reports like dataview is shown, and arrow keys cannot be used to select Crowdin.

**Comment by nvdaes** [comment]
Thanks @vslavik . Hope now my description is more useful.

**Comment by vslavik** [comment]
You assume, wrongly, that I am familiar with NVDA and can use it and that it is easy to reproduce your specific environment. If you struggle with describing the problem, but can reproduce it, could you submit a PR instead?

> A control accessible with keyboard, such as a combo box

I don't know why you think there should be a "combo box" there...?

> and arrow keys cannot be used to select Crowdin.

It is unclear from your description why you think that? Dataview controls are navigable with keyboard and Crowdin is the first selection, so already selected: self-evidently, it *can* be selected. There's some crucial bit of information, which is probably obvious to you, missing.

**Comment by nvdaes** [comment]
> I don't know why you think there should be a "combo box" there...?

Since in combo boxes NVDA reads properly what is selected, for example in the combo box used to select between projects stored in Crowdin.

If you think that here's better to use a dataview, then I'll report this to NVDA in case it can be fixed there. Otherwise, I'll try to create a pull request in Poedit, probably trying to add a combo box.

**Comment by vslavik** [comment]
> Since in combo boxes NVDA reads properly what is selected

So I guess we should change all of Poedit to be only combo boxes and nothing but combo boxes then? Again, dataview controls *are* a11y-visible and *are* keyboard navigable, contrary to the plain reading of what you wrote.

> If you think that here's better to use a dataview, then I'll report this to NVDA in case it can be fixed there.

Again, I don't even understand why you think there's anything wrong and *what* do you think is wrong. Your first step should be to coherently describe the problem in enough detail that a person-that-is-not-you can understand it; see the article linked in the instructions.

> Otherwise, I'll try to create a pull request in Poedit, probably trying to add a combo box.

*That* I am certainly not going to accept.

**Comment by CyrilleB79** [comment]
@nvdaes wrote:

> A control that NVDA screen reader reports like dataview is shown, and arrow keys cannot be used to select Crowdin.

Actually arrow keys can help change the selection in the list. One part of the problem is that as soon as you change the selected item in the list with up/down arrow, the focus is moved by poedit to the Connect button. This makes the exploration of the items of this list more difficult and less predictable for a blind person. Also, today, there are only 2 items in this list; but in case there is a third item one day, it would make keyboard navigation more difficult (even for a sighted person).

@vslavik is this focus jump triggered by selection change an intended design choice. If yes, would it be acceptable to remove it so that it is guaranteed that using keyboard navigation (up/down arrows) in the list, the focus remain in that list?

There are also other accessibility issues in this dialog with controls that are not correctly reported by NVDA nor with Narrator. But for now, I am not able to precisely point where the issue lies.

**Comment by vslavik** [comment]
> Hope explanation provided by @CyrilleB79 is clear to you.

Yes, although its relationship to your *very* different claims remains a mystery :(

> Actually arrow keys can help change the selection in the list. One part of the problem is that as soon as you change the selected item in the list with up/down arrow, the focus is moved by poedit to the Connect button.

That's weird. I can reproduce it, and also in another place in the Settings window: if you use keyboard navigation through the notebook tabs, as soon as you arrive at the "Accounts" tab, focus is stolen too. I.e. it seems unrelated to the list per se.

> Also, today, there are only 2 items in this list; but in case there is a third item one day, it would make keyboard navigation more difficult (even for a sighted person).

Sorry, you lost me again. How so? The list's keyboard navigation is the same if it has 2 or 200 items, no?

> @vslavik is this focus jump triggered by selection change an intended design choice.

No.

**Comment by CyrilleB79** [comment]
> > Hope explanation provided by [@CyrilleB79](https://github.com/CyrilleB79) is clear to you.
> 
> Yes, although its relationship to your _very_ different claims remains a mystery :(

No mystery here. @nvdaes and myself are both blind and NVDA users. We have both experienced the same difficulty in the cloud connection dialog.

From this experience, @nvdaes has assumed that the list was not accessible, whereas I have understood that it was partially, but not easily for screen users because the focus jump did not allow us to understand this dialog at first glance.

> > Also, today, there are only 2 items in this list; but in case there is a third item one day, it would make keyboard navigation more difficult (even for a sighted person).
> 
> Sorry, you lost me again. How so? The list's keyboard navigation is the same if it has 2 or 200 items, no?

Because if you use arrows to select another item in the list:
* with 2 items, you arrow once, then the focus jumps out of the list but you have been able to select the correct item.
* with 3 or more items, you may need to up or down arrow more than one time in the list. Due to the focus jump, you'll need to shif+tab back in the list after each arrow press until you select the wished item.

Anyway, if it's not clear, no matter, that's a detail. In any case, the focus should remain in the list.
